### DISCUSSION TITLE
How to clear input field without removing the data from form state

### DISCUSSION THREAD

**Discussion by pini85** [react-hook-form/react-hook-form#8894] [Q&A]
In a certain input field I am adding values to an array which is stored in the react hooks form state. What I would like to do is each time I save the value, I want to clear the input field, focus on it again so the user can type in more values.
I tried using reset but unfortunately it removes the values from state. I want to only clear the input field. Any suggestions?

**Comment by Moshyfawn** [comment]
I'm not entirely sure what your use-case is, but you to keep your form state but reset the field values with the `reset` from method, you might wanna look into the `reset` options [like `keepDirty`, `keepIsSubmitted`](https://react-hook-form.com/api/useform/reset), etc.

**Reply by pini85** [reply]
@Moshyfawn 
Here is a snippet to illustrate better what I want to do. 
Each time I will add a value from the input, I would like the input field to be empty. When I submit the form it should send the array with it, but the reset overrides the form values.
Would appreciate it any help! 
Thanks
https://codesandbox.io/s/compassionate-bird-jxp8wc?file=/src/App.js
